dbt/adapters/spark_cde/cdeapisession.py

```diff
@@ -119,24 +119,34 @@ def execute(self, sql: str, *parameters: Any) -> None:
     
         # 3. run the job
         job_status = self._cde_connection.getJobRunStatus(job).json()
-        
-        # 4. wait for the result       
-        while job_status["status"] != CDEApiConnection.JOB_STATUS['succeeded']:
+
+        # 4. wait for the result
+        while (job_status["status"] != CDEApiConnection.JOB_STATUS['succeeded']) or (job_status["status"] == CDEApiConnection.JOB_STATUS['failed']):
```

Comment:
since it throws exception for failed, while condition for failed will not be reached.

Reply:
Good find. I will do the negative test scenario(job status failed) and update PR.

Reply:
Updated. i reverted my flawed logic.  Testing takes a lot of time though as you saw from a sample bms session. Will  refactor/cleanup the code in future and not this pr.